File: arrays/boats-to-save-people/boats-to-save-people.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
function numRescueBoats(people, limit) {
  people.sort((a, b) => a - b);

  let i = 0;
  let j = people.length - 1;
  let boats = 0;

  while (i <= j) {
    boats++;

    if (people[i] + people[j] <= limit) {
      i++;
    }

    j--;
  }

  return boats;
}

File: arrays/boats-to-save-people/boats-to-save-people.go
package boatstosavepeople

import "sort"

func numRescueBoats(people []int, limit int) int {
	sort.Ints(people)
	i := 0
	j := len(people) - 1
	boats := 0

	for i <= j {
		boats++
		if people[i]+people[j] <= limit {
			i++
		}
		j--
	}

	return boats
}
